code/service/src/main/scala/com/namely/chiefofstate/StartNodeBehaviour.scala

```diff
@@ -6,224 +6,65 @@
- *   <li> run the required schemas and migration needed
- *   <li> the telemetry tools and the various gRPC interceptors
- *   <li> the gRPC service
- * </ul>
- */
 object StartNodeBehaviour {
   final val log: Logger = LoggerFactory.getLogger(getClass)
 
   def apply(config: Config): Behavior[NotUsed] = {
     Behaviors.setup { context =>
-      // get the  COS config
-      val cosConfig: CosConfig = CosConfig(config)
 
       val cluster: Cluster = Cluster(context.system)
       context.log.info(s"starting node with roles: ${cluster.selfMember.roles}")
 
       // Start the akka cluster management tool
       AkkaManagement(context.system).start()
+      // start the cluster boostrap
       ClusterBootstrap(context.system).start()
 
-      // create and run the migrator
-      val migrator: Migrator = {
-        val journalJdbcConfig: DatabaseConfig[JdbcProfile] =
-          JdbcConfig.journalConfig(config)
-
-        // FIXME: this uses the write-side slick config to read the journal
-        // for the migration, as COS projections has moved to raw JDBC
-        // connections and no longer uses slick.
-        val projectionJdbcConfig: DatabaseConfig[JdbcProfile] =
-          JdbcConfig.projectionConfig(config, "write-side-slick")
-
-        // TODO: think about a smarter constructor for the migrator
-        // get the projection config
-        val priorProjectionJdbcConfig: DatabaseConfig[JdbcProfile] =
-          JdbcConfig.projectionConfig(config, "chiefofstate.migration.v1.slick")
-
-        // get the old table name
-        val priorOffsetStoreName: String = config.getString("chiefofstate.migration.v1.slick.offset-store.table")
-
-        val v1: V1 = V1(journalJdbcConfig, priorProjectionJdbcConfig, priorOffsetStoreName)
-        val v2: V2 = V2(journalJdbcConfig, projectionJdbcConfig)(context.system)
-        val v3: V3 = V3(journalJdbcConfig)
-
-        new Migrator(journalJdbcConfig)
-          .addVersion(v1)
-          .addVersion(v2)
-          .addVersion(v3)
-      }
-
-      migrator.run() match {
-        case Failure(exception) => throw exception // we want to panic here.
-        case Success(_) => // We only proceed when the data stores and various migrations are done successfully.
-          log.info("ChiefOfState migration successfully done. About to start...")
-      }
-
-      // start the telemetry tools and register global tracer
-      TelemetryTools(cosConfig).start()
-
-      // We only proceed when the data stores and various migrations are done successfully.
-      log.info("Journal and snapshot store created successfully. About to start...")
-
-      val channel: ManagedChannel =
-        NettyHelper
-          .builder(
-            cosConfig.writeSideConfig.host,
-            cosConfig.writeSideConfig.port,
-            cosConfig.writeSideConfig.useTls
+      // initialize the service bootstrapper
+      val bootstrapper: ActorRef[StartCommand] =
+        context.spawn(
```

Comment:
as it is now, this will start the service before the migration has run. I believe this will actually fail for both new apps and migrations.